internal/controller/networkpolicy/np_objects.go

```diff
@@ -95,6 +133,53 @@ func buildMainNetworkPolicy(desired *flowslatest.FlowCollector, mgr *manager.Man
 				Port:     ptr.To(intstr.FromInt32(constants.WebhookPort)),
 			}},
 		})
+		// Allow host
+		np.Spec.Egress = append(np.Spec.Egress, networkingv1.NetworkPolicyEgressRule{
+			To: []networkingv1.NetworkPolicyPeer{},
+			Ports: []networkingv1.NetworkPolicyPort{{
+				Protocol: ptr.To(corev1.ProtocolTCP),
+				Port:     ptr.To(intstr.FromInt32(constants.K8sAPIServerPort)),
+			}},
+		})
+		// Allow apiserver
+		np.Spec.Ingress = append(np.Spec.Ingress, networkingv1.NetworkPolicyIngressRule{
+			From: []networkingv1.NetworkPolicyPeer{
+				{
+					NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						"policy-group.network.openshift.io/host-network": "",
+					}},
+				},
+			},
+			Ports: []networkingv1.NetworkPolicyPort{{
+				Protocol: ptr.To(corev1.ProtocolTCP),
+				Port:     ptr.To(intstr.FromInt32(constants.WebhookPort)),
+			}},
+		})
+		np.Spec.Egress = append(np.Spec.Egress, networkingv1.NetworkPolicyEgressRule{
+			To: []networkingv1.NetworkPolicyPeer{
+				peerInNamespace(constants.DNSNamespace),
+			},
+		})
+		// Allow egress for FLP when some exporters are configured
+		if len(desired.Spec.Exporters) > 0 {
+			np.Spec.Egress = append(np.Spec.Egress, networkingv1.NetworkPolicyEgressRule{
+				To: []networkingv1.NetworkPolicyPeer{
+					{
+						NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
```

Comment:
I'm not sure to understand this rule, it says traffic allowed to namespace selector "app": "flowlogs-pipeline" ? Which namespace is that? Is it in order to allow users to configure namespaces where they deployed their custom exporter?

Reply:
Thanks for catching this, I did some cleaning after testing and incorectly changed this...

The goal is to allow every egress for FLP only if an exporter is configured.

Reply:
ah, I see ... else, there's also the AdditionalNamespace that can be used for that purpose, no?

Reply:
Some exporters may target something outside of the Openshift cluster. We have two possibilities here :
- allow every egress traffic for FLP if an exporter is configured
- do nothing and document how to configure an aditional NP when also configuring an exporter
I was going for the first option but I am also fine with the second